pkg/v1/tkg/client/machine_deployment.go

```diff
@@ -108,6 +108,15 @@ func (c *TkgClient) SetMachineDeployment(options *SetMachineDeploymentOptions) e
 		}
 		return c.SetNodePoolsForPacificCluster(clusterClient, options)
 	}
+
+	var cluster capi.Cluster
+	if err = clusterClient.GetResource(&cluster, options.ClusterName, options.Namespace, nil, nil); err != nil {
+		return errors.Wrap(err, "Unable to retrieve cluster resource")
+	}
+	if cluster.Spec.Topology.Class != "" {
```

Comment:
`cluster.Spec.Topology` may be `nil`. If it is, this will panic.

Reply:
In fact, it may actually be sufficient to test if `cluster.Spec.Topology != nil` here.